lib/tuples.gi

```diff
@@ -139,26 +139,47 @@ InstallOtherMethod( DirectProductElement,
     return DirectProductElementNC( fam, objlist );
     end );
 
+#############################################################################
+##
+#M  String( <dpelm> )  . . . . . . convert a direct product element to string
+##
+InstallMethod( String,
+    "for a direct product element",
+    [ IsDirectProductElement ],
+    function( dpelm )
+    local i, str;
+    str := "DirectProductElement( [ ";
+    for i in [ 1 .. Length( dpelm ) ] do
+      if i > 1 then
+        Append(str, ", ");
+      fi;
+      Append(str, String( dpelm[i] ));
+    od;
+    Append(str, " ] )");
+    return str;
+    end );
 
 #############################################################################
 ##
-#M  PrintObj( <dpelm> )  . . . . . . . . . . . print a direct product element
+#M  PrintString( <dpelm> ) . convert a direct product element to print string
 ##
-InstallMethod( PrintObj,
+InstallMethod( PrintString,
     "for a direct product element",
     [ IsDirectProductElement ],
     function( dpelm )
-    local i;
-    Print( "DirectProductElement( [ " );
-    for i in [ 1 .. Length( dpelm )-1 ] do
-      Print( dpelm[i], ", " );
+    local i, str;
+    str := "DirectProductElement( [ ";
+    for i in [ 1 .. Length( dpelm ) ] do
+      if i > 1 then
```

Comment:
Performing this check in each loop iteration is less efficient than what the old code did.